axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/PurchaseOrderServiceSupplychainImpl.java

```diff
@@ -365,4 +374,95 @@ public void updateToValidatedStatus(PurchaseOrder purchaseOrder) {
     purchaseOrder.setStatusSelect(PurchaseOrderRepository.STATUS_VALIDATED);
     purchaseOrderRepo.save(purchaseOrder);
   }
+
+  @Override
+  public String createShipmentCostLine(PurchaseOrder purchaseOrder) throws AxelorException {
+    PurchaseOrderService purchaseOrderService = Beans.get(PurchaseOrderService.class);
```

Comment:
I think you can use `this` instead of injecting purchase order service here since a refactor